File: backend/app/schemas/results.py
from pydantic import BaseModel
from typing import List

class ResultItem(BaseModel):
    key: str
    label: str
    participant: str
    answer_key: str
    status: str  # "correct" | "wrong"

class ResultBlock(BaseModel):
    key: str
    label: str
    items: List[ResultItem]

class EvaluationResultResponse(BaseModel):
    round_id: str
    blocks: List[ResultBlock]

File: frontend/src/types/results.ts
// src/types/results.ts

export interface ResultItem {
  key: string;
  label: string;
  participant: string;
  answer_key: string;
  status: "correct" | "wrong" | "partial"; // inclui parcial
}

export interface ResultBlock {
  key: string;
  label: string;
  items: ResultItem[];
}

export interface EvaluationResultResponse {
  round_id: string;
  blocks: ResultBlock[];
}
